src/renderers/shared/stack/event/ReactSyntheticEventType.js

```diff
@@ -8,14 +8,29 @@
  *
  * Flow type for SyntheticEvent class that includes private properties
  *
- * @providesModule ReactSyntheticEvent
+ * @providesModule ReactSyntheticEventType
  * @flow
  */
 
 'use strict';
 
-export type DispatchConfig = any;
+import type {ReactInstance} from 'ReactInstanceType';
+
+export type DispatchConfig = {
+  dependencies: Array<string>,
+  phasedRegistrationNames?: {
+    bubbled: string,
+    captured: string,
+  },
+  registrationName?: string,
+};
 
 export class ReactSyntheticEvent extends SyntheticEvent {
   dispatchConfig: DispatchConfig;
+  getPooled: (
```

Comment:
https://github.com/facebook/react/blob/a8741963dc661432caa906c295a0477691bad614/src/renderers/shared/stack/event/SyntheticEvent.js#L258-L260

If you type SyntheticEvent in this file, and replace the exports to be

```js
module.exports = PooledClass.addPoolingTo(SyntheticEvent, PooledClass.fourArgumentPooler);
```

then it'll automatically get `getPooled` and `release` static methods because `PooledClass` has been typed. The only caveat is that I didn't find a way to get the correct type of getPooled arguments which are the constructor arguments (if you find a way that would be awesome).